AllTheThings.lua

```diff
@@ -16746,6 +16746,53 @@ RowOnEnter = function (self)
 		if reference.speciesID then
 			local progress, total = C_PetJournal.GetNumCollectedInfo(reference.speciesID);
 			if total then GameTooltip:AddLine(tostring(progress) .. " / " .. tostring(total) .. L["COLLECTED_STRING"]); end
+			
+			if progress > 0 and app.Settings:GetTooltipSetting("PetBattleQuality") then
+				local speciesName = C_PetJournal.GetPetInfoBySpeciesID(reference.speciesID);
+				
+				-- Set the journal search filter to the pet name and the category for example duskwood or in-game shop
+				-- This is done so we limit how many pets will show up.
+				-- If category text is formatted we only search for name since the formatting will break the search
+				if string.find(reference.parent.parent.text, "|") == nil then
+					C_PetJournal.SetSearchFilter(speciesName .. " " .. reference.parent.parent.text);
+				else
+					C_PetJournal.SetSearchFilter(speciesName);
+				end
+				-- C_PetJournal.SetSearchFilter(speciesName .. " " .. reference.parent.parent.text);
+				local quality = {[1] = 0, [2] = 0, [3] = 0, [4] = 0};
+				local petlevel = {[1] = " ", [2] = " ", [3] = " ", [4] = " "}
+
+				-- Upper limit to serach for is 100 to make sure it wont run forever
+				-- it will stop running once theres no more collected pets left in journal which is normally under 10 
+				for i = 1, 100, 1 do
+					local guid, _, _, _, level, _, _, journalName = C_PetJournal.GetPetInfoByIndex(i);
+					if guid == nil then
+						if i == 1 then
+							-- Some pets are on the line of map borders eg Small Frog
+							-- This pet is not listed as spawning in duskwood in the wow sources, cause its on the westfall-duskwood border
+							-- For theese edge cases we change the serach to just the species name.
```

Comment:
Oh... we're expecting that `.parent.parent` always corresponds to a *Zone Name*? Hmmm... this is definitely not always true 🤔 

Is there no way to pull this info from the Pet Journal directly from the SpeciesID? I haven't looked into that...

Reply:
This also works with dungeons and raids, the search works with sources. So black temple will for example also show pets that drop from there.

Details about collected pet comes from C_PetJournal and i could not find a way to reference SpeciesID with C_PetJournal
It works with indexes in the journal list, searching for the speciesID also doesn't work.

Reply:
Alright well it seems fine for now. I see some potential logic clean up for nit-pick things, but no reason to hold off the PR for that. Can address it at some later time when I find time. I will pull this branch down tonight and just double check in game myself before merging it.